src/main.cc

```diff
@@ -126,28 +126,14 @@ tarantool_uptime(void)
 	return ev_monotonic_now(loop()) - start_time;
 }
 
-/**
- * Create a checkpoint from signal handler (SIGUSR1)
- */
-static int
-sig_checkpoint_f(va_list ap)
-{
-	(void)ap;
-	if (box_checkpoint() != 0)
-		diag_log();
-	return 0;
-}
-
 static void
 sig_checkpoint(ev_loop * /* loop */, struct ev_signal * /* w */,
 	     int /* revents */)
 {
-	struct fiber *f = fiber_new_system("checkpoint", sig_checkpoint_f);
-	if (f == NULL) {
-		say_warn("failed to allocate checkpoint fiber");
+	say_info("got SIGUSR1 signal bound to checkpoint");
+	if (shutdown_started)
```

Comment:
I'd introduce the flag `is_storage_shutdown` and check it in `box_checkpoint_async`.

Reply:
Done. While at it let's also make `sig_checkpoint` naming, indentation and handling of unused arguments on par with other signal functions.

<details>
<summary>diff</summary>


```diff
diff --git a/src/box/box.cc b/src/box/box.cc
index 316cfe39c..0304c62d7 100644
--- a/src/box/box.cc
+++ b/src/box/box.cc
@@ -154,6 +154,8 @@ static int box_read_ffi_disable_count;
  */
 static bool is_box_configured = false;
 static bool is_storage_initialized = false;
+/** Set if storage shutdown is started. */
+static bool is_storage_shutdown = false;
 static bool is_ro = true;
 static fiber_cond ro_cond;
 
@@ -5606,13 +5608,14 @@ box_cfg(void)
 void
 box_checkpoint_async(void)
 {
-	if (!is_storage_initialized)
+	if (!is_storage_initialized || is_storage_shutdown)
 		return;
 	gc_trigger_checkpoint();
 }
@@ -5972,6 +5975,7 @@ box_storage_shutdown()
 {
 	if (!is_storage_initialized)
 		return;
+	is_storage_shutdown = true;
 	iproto_shutdown();
 	box_watcher_shutdown();
 	/*
diff --git a/src/main.cc b/src/main.cc
index 1901b3d8a..714eaa3a9 100644
--- a/src/main.cc
+++ b/src/main.cc
@@ -127,12 +127,12 @@ tarantool_uptime(void)
 }
 
 static void
-sig_checkpoint(ev_loop * /* loop */, struct ev_signal * /* w */,
-	     int /* revents */)
+signal_checkpoint(ev_loop *loop, struct ev_signal *w, int revents)
 {
+	(void)loop;
+	(void)w;
+	(void)revents;
 	say_info("got SIGUSR1 signal bound to checkpoint");
-	if (shutdown_started)
-		return;
 	box_checkpoint_async();
 }
 
@@ -281,7 +281,7 @@ signal_init(void)
 	fiber_signal_init();
 	crash_signal_init();
 
-	ev_signal_init(&ev_sigs[0], sig_checkpoint, SIGUSR1);
+	ev_signal_init(&ev_sigs[0], signal_checkpoint, SIGUSR1);
 	ev_signal_init(&ev_sigs[1], signal_sigint_cb, SIGINT);
 	ev_signal_init(&ev_sigs[2], signal_cb, SIGTERM);
 	ev_signal_init(&ev_sigs[3], signal_sigwinch_cb, SIGWINCH);

```

</details>